app/use_cases/petition_plugin/detail_updater.rb

```diff
@@ -0,0 +1,14 @@
+module PetitionPlugin
+  class DetailUpdater
+    def perform(detail, attrs, version_attrs)
+      detail.update_attributes attrs 
+
+      current_version = detail.current_version
+      if current_version.nil? || current_version.body != version_attrs[:body]
+        detail.petition_detail_versions << PetitionPlugin::DetailVersion.new(version_attrs)
+      end
+
+      detail.save
```

Comment:
The only downside in returning this is that if you need the `detail.errors` you will need to use the reference you pass to `perform`. Though this will work, it is not optimal, and if you wrote the controller tests you would notice this.

Nevertheless, I don't think you're currently using the errors. Are there validations?